plugins/outputs/wavefront/wavefront.go

```diff
@@ -173,7 +173,8 @@ func (w *Wavefront) Write(metrics []telegraf.Metric) error {
 			err := w.sender.SendMetric(point.Metric, point.Value, point.Timestamp, point.Source, point.Tags)
 			if err != nil {
 				if isRetryable(err) {
-					return fmt.Errorf("wavefront sending error: %v", err)
+					w.Log.Errorf("wavefront sending error: %v", err)
+					return w.sender.Flush()
```

Comment:
To be honest I would prefer
```suggestion
					if flushErr := w.sender.Flush(); flushErr != nil {
						w.Log.Errorf("wavefront flushing error: %v", flushErr)
					}
					return err
```
as the sending error is the relevant one to return IMO. What do you think?